### PR TITLE
Refactor some of TryToInterpolate.hpp

### PR DESCRIPTION
**Pull request by markscheel**
## Proposed changes

Move some of interpolate_data and small functions into new functions
in InterpolateTargetDetail.

This is so in my next PR I can use those functions in
InterpolateWithoutInterpComponent without duplicating code.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp [hunk 1]
```diff
@@ -604,5 +605,72 @@ void set_up_interpolation(
...
+template <typename InterpolationTargetTag, typename SourceTags,
+          typename Metavariables, typename ElementId>
+void compute_dest_vars_from_source_vars(
+    gsl::not_null<Variables<
+        typename InterpolationTargetTag::vars_to_interpolate_to_target>*>
+        dest_vars,
+    const Variables<SourceTags>& source_vars,
+    const Domain<Metavariables::volume_dim>& domain,
+    const Mesh<Metavariables::volume_dim>& mesh, const ElementId& element_id,
+    Parallel::GlobalCache<Metavariables>& cache,
```
Comment (nilsdeppe) [inline src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp]: I think the `cache` can be `const` if you are only retrieving items from it

File: src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp [hunk 2]
```diff
@@ -604,5 +605,72 @@ void set_up_interpolation(
...
+    tmpl::any<TagList, tmpl::bind<detail::any_index_in_frame_impl, tmpl::_1,
+                                  Frame>>::value;
+
+/// Calls compute_vars_to_interpolate to compute
+/// InterpolationTargetTag::vars_to_interpolate_to_target from the source
+/// variables.  Does any frame tranformations needed.
+template <typename InterpolationTargetTag, typename SourceTags,
+          typename Metavariables, typename ElementId>
+void compute_dest_vars_from_source_vars(
+    gsl::not_null<Variables<
```
Comment (nilsdeppe) [inline src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp]: `const gsl::not_null` (add `const`)

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
LGTM, two minor requests. Please squash immediately!

File: src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp [hunk 3]
```diff
@@ -604,5 +605,72 @@ void set_up_interpolation(
...
       });
 }
+
+CREATE_HAS_TYPE_ALIAS(compute_vars_to_interpolate)
+CREATE_HAS_TYPE_ALIAS_V(compute_vars_to_interpolate)
+
+namespace detail {
+template <typename Tag, typename Frame>
+using any_index_in_frame_impl =
+    TensorMetafunctions::any_index_in_frame<typename Tag::type, Frame>;
```
Comment (knelli2) [inline src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp:616]: `#include DataStructures/Tensor/Metafunctions.hpp`

**Review by knelli2 — CHANGES_REQUESTED** [review]
Also squash immediately.

**Review by knelli2 — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**